src/core/engine/audiorenderer.cpp

```diff
@@ -379,6 +386,38 @@ void AudioRenderer::updateInterval()
     m_writeInterval = interval;
 }
 
+void AudioRenderer::calculateGain()
+{
+    m_gainScale = 1.0;
+
+    if(!m_settings->value<Settings::Core::ReplayGainEnabled>()) {
+        return;
+    }
+
+    float peak{0.0};
+    const auto gainType = static_cast<ReplayGainType>(m_settings->value<Settings::Core::ReplayGainType>());
+    switch(gainType) {
+        case(ReplayGainType::Track):
+            m_gainScale = std::pow(10.0, m_currentTrack.replayGainTrackGain() / 20.0);
+            peak        = m_currentTrack.replayGainTrackPeak();
+            break;
+        case(ReplayGainType::Album):
+            m_gainScale = std::pow(10.0, m_currentTrack.replayGainAlbumGain() / 20.0);
+            peak        = m_currentTrack.replayGainAlbumPeak();
+            break;
+    }
+
+    const auto preamp = static_cast<float>(m_settings->value<Settings::Core::ReplayGainPreAmp>());
+    m_gainScale *= std::pow(10.0, preamp / 20.0);
```

Comment:
We can add preamp earlier (in the first pow) and reduce to

`m_gainScale = std::pow(10.0, (m_currentTrack.replayGain{Album,Track}Gain() + preamp) / 20.0)`

to save the CPU some math work

Reply:
Good call. It shouldn't matter too much as the method is only called on track change/changing settings, but it does simplify it a bit.